CI `Rust / cargo examples (amd64) (pull_request)` failing on branch-50

### Describe the bug

This CI check is failing on the branch-50 for some reason. Here is an example

https://github.com/apache/datafusion/actions/runs/18597927327/job/53029362401?pr=18131

Looks to me like it ran out of disk space

```
error: could not compile `datafusion-examples` (example "flight_client") due to 1 previous error
error: linking with `cc` failed: exit status: 1
  |
  = note:  "cc" "-m64" "/tmp/rustc2UPVcs/symbols.o" "<78 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "/__w/datafusion/datafusion/target/ci/deps/{libdatafusion-2aba99b464d514e4.rlib,libdatafusion_functions_table-c7d656b38778fc8a.rlib,libdatafusion_functions_window-b56bf6d8548383e7.rlib,libdatafusion_functions_nested-b297acb365e010f1.rlib,libdatafusion_physical_optimizer-5a1b0d304906fabb.rlib,libdatafusion_optimizer-ae855e0fbf9fe212.rlib,libdatafusion_catalog_listing-c938274061f725be.rlib,libdatafusion_datasource_parquet-7dcbe66a019fd3a9.rlib,libdatafusion_pruning-fe3f4a26d68b2df2.rlib,libdatafusion_datasource_json-1d593650ce43611d.rlib,libdatafusion_datasource_csv-b349fcb2db5a3147.rlib,libdatafusion_sql-07d3f209cab0a8f1.rlib,libbigdecimal-98642724cb028a2d.rlib,liblibm-1a98895e1ac6b7e5.rlib,libdatafusion_catalog-45f20d200311f716.rlib,libdatafusion_datasource-526c45c63b747213.rlib,libglob-1531ab0f9e0f472f.rlib,libdatafusion_physical_expr_adapter-0298a7943f616d2b.rlib,libdatafusion_session-7b0ad3b9abe3701f.rlib,libbzip2-aa0890b94fcaddc5.rlib,liblibbz2_rs_sys-6588f0366fef2608.rlib,libasync_compression-2622be4ff1c66a22.rlib,libxz2-76b67708433af602.rlib,liblzma_sys-90188c933fc77fe1.rlib,libbzip2-06a3f5729882b62d.rlib,libbzip2_sys-307bafe6d7634420.rlib,libdatafusion_physical_plan-32b83dd87a98ba48.rlib,libdatafusion_common_runtime-f0880f8fd8d5ce79.rlib,libdatafusion_functions_aggregate-b406f73a6decd152.rlib,libdatafusion_physical_expr-2dd9173ad720211b.rlib,libpetgraph-ecad1e3ad1abe1b8.rlib,libfixedbitset-668ffa83d11221aa.rlib,libdatafusion_functions-5186708a2402cfa5.rlib,libsha2-d7550e2e8f68fcb7.rlib,libcpufeatures-35d273103b04283f.rlib,libblake3-d41f49a11c2de6ca.rlib,libconstant_time_eq-e3cc299489ab04d1.rlib,libarrayvec-0e50991b89522b64.rlib,libarrayref-b076c31b5836aca2.rlib,libblake2-6965b9923ca68cb0.rlib,libuuid-aa7cb4ad324388a9.rlib,libdatafusion_execution-e3a4595854ffc567.rlib,libtempfile-8b5b25c406162cb4.rlib,libfastrand-672854325d4ce41b.rlib,librustix-338e766feedfc123.rlib,liblinux_raw_sys-6f1dfbad544ac87f.rlib,libdashmap-b03e167ea8b2e34e.rlib,libcrossbeam_utils-b71fe6153acbfdf7.rlib,libdatafusion_expr-2a603599f7a78ef1.rlib,libdatafusion_functions_aggregate_common-c8d3994650a6edfd.rlib,libdatafusion_functions_window_common-0da1e9712cce78ad.rlib,libdatafusion_doc-caf470e90b9dea0c.rlib,libdatafusion_physical_expr_common-a4a6a5c8aedeba9b.rlib,libdatafusion_expr_common-933ae0dc827f3882.rlib,libdatafusion_common-cdd808263897a79f.rlib,libhashbrown-04a6619f6915d14a.rlib,libparquet-85df72b50ef59561.rlib,libbrotli-9656958e33816406.rlib,libbrotli_decompressor-9b7325719dba90ff.rlib,liballoc_stdlib-b45a7be5eccaa0b6.rlib,liballoc_no_stdlib-8c66105e5db757ed.rlib,libflate2-d8f7d40e4c5a5ebf.rlib,libminiz_oxide-1cdd9dc245b1c0f2.rlib,libadler2-23b86448012b97bc.rlib,liblibz_rs_sys-d0008adc81287544.rlib,libzlib_rs-2485c0ec96f77f03.rlib,libcrc32fast-11e3985fd3cabb13.rlib,libsnap-cbf357189c36ff93.rlib,libobject_store-cbcfc4eea4ffa231.rlib,libhumantime-a16e98c3b4a42d03.rlib,libthiserror-af3e395179723ea2.rlib,librand-11a0cde927916533.rlib,librand_chacha-65abe72671e3a922.rlib,libppv_lite86-bd8bfc75aa82c452.rlib,librand_core-c020d36c9ad0ca1b.rlib,libwalkdir-ed902a35fef3b93d.rlib,libsame_file-a00cdd42b1a7ea0b.rlib,libquick_xml-b63eafdccc747de6.rlib,libmd5-e0cb0e81909cedde.rlib,libdigest-33e2948434ba6f9a.rlib,libblock_buffer-81cfbcb9d69188d5.rlib,libcrypto_common-916ac78576671ecf.rlib,libgeneric_array-56a3f676a3427cdb.rlib,libtypenum-20bedc432310c282.rlib,libitertools-f97e8c174cbb6ee6.rlib,libeither-14c1c48e853712e5.rlib,libreqwest-a0caba57c57d8d45.rlib,libserde_urlencoded-c7f0bcf48e3e395e.rlib,libhyper_rustls-fb4da0058a7f5d95.rlib,librustls_native_certs-22f3d3787c8165ec.rlib,libopenssl_probe-5ebdcf6485137f9a.rlib,libtokio_rustls-480a5feef19761f0.rlib,librustls-ae21bded921f1694.rlib,libsubtle-31c3a8c077c295d7.rlib,libwebpki-c1071ef6e4ff4d21.rlib,libring-84f8e63b9c9a697f.rlib,libgetrandom-4fcbc295b4d518c6.rlib,libuntrusted-7d7d757fde335605.rlib,librustls_pki_types-3081fc2cdcafaa1c.rlib,libzeroize-fd9147ae4fff9cd0.rlib,libtower_http-68c0e826f2754d3a.rlib,libiri_string-37841017961071f1.rlib,libtower-9d8360be85772b5c.rlib,libtower_layer-0c93bdc4dff0ab23.rlib,libhyper_util-695cfdbbed6d428b.rlib,libipnet-1d0a57e0afed1b9b.rlib,libtower_service-1e050a7a6fd6726f.rlib,libhyper-82d527e226c85c7e.rlib,libwant-05029c4a510d7db5.rlib,libtry_lock-02e96cc8294eed2c.rlib,libhttparse-48aee97b1ab5a817.rlib,libh2-529437dd8dd1dd91.rlib,libtracing-976b2c2fd23f0f0a.rlib,libtracing_core-9ceeaa2c2ff8c8b6.rlib,libatomic_waker-1f53812e9e070841.rlib,libhttpdate-b62245b3a73f8539.rlib,libtokio_util-2650fc74cf09b039.rlib,libhttp_body_util-fc4726bd5a36d509.rlib,libhttp_body-8ad858d35564033d.rlib,liburl-733b98d83f10d858.rlib,libidna-8a8d3cd0b00dda30.rlib,libutf8_iter-3786730b9144ad3e.rlib,libidna_adapter-d57494fe1854b0a3.rlib,libicu_properties-e1450731fff23f9e.rlib,libicu_properties_data-15e409e29be745e1.rlib,libicu_normalizer-acb19b86fb0484d8.rlib,libicu_normalizer_data-ba88bca505d4debd.rlib,libicu_collections-6989d9572ad8a864.rlib,libpotential_utf-7c124663aae7aacb.rlib,libicu_provider-0235aa0d253fce1f.rlib,libicu_locale_core-a3a3992fb202b6dc.rlib,libtinystr-e8d03a0f006e9735.rlib,liblitemap-e3e729224945a3c7.rlib,libwriteable-a52a3816d9f0dfcd.rlib,libzerovec-49d470671ed68dda.rlib,libzerotrie-9191cf999086c0bf.rlib,libyoke-2a754070d7ad4c1e.rlib,libstable_deref_trait-c845a28d080d4452.rlib,libzerofrom-a2dcb00e2db9aa2c.rlib,libform_urlencoded-d4ebe2caa44dc3a2.rlib,libpercent_encoding-735f542fa8ad06cb.rlib,libhttp-05c9cf7a6ca6734c.rlib,libfnv-5de475a82d9b1ca9.rlib,libsync_wrapper-65013f5df38318f8.rlib,libtokio-27ff40b4fbff9313.rlib,libsocket2-b3f3ede022a56bca.rlib,libmio-c7cb5d5c0ba0aa9c.rlib,libparking_lot-7066c8c606e88182.rlib,libparking_lot_core-1529457563ed481b.rlib,libsmallvec-5b89fefdb7c5c962.rlib,liblock_api-16f85892576e7add.rlib,libscopeguard-5c873fa36351ddb4.rlib,libfutures-afbb915af87d28d8.rlib,libfutures_executor-d8e79a0d2a050fde.rlib,libfutures_util-07cfe98defd91424.rlib,libfutures_io-7289aba39c755de2.rlib,libslab-72830ea603a912da.rlib,libfutures_channel-8984bff1e45abbaa.rlib,libpin_project_lite-2d17a11d7198541a.rlib,libfutures_sink-7cf1eaab0a8b9d53.rlib,libfutures_task-d8c461c420a8da33.rlib,libpin_utils-fdce69db4b701e31.rlib,libfutures_core-0fc5ec3f6c128695.rlib,libthrift-b71ea64fd42ef7bb.rlib,libordered_float-d311cd8c94867032.rlib,libinteger_encoding-374dcc752fc43f33.rlib,libbyteorder-bc7a84cde457560f.rlib,libhex-b52dc89723108cff.rlib,libsqlparser-290e7ee88613a1a0.rlib,librecursive-8570ae8f83d79ef9.rlib,libstacker-1791b45c2d04f1a2.rlib,libpsm-b41ffc90c6d5631d.rlib,liblog-c77da54b01ca4cd6.rlib,libarrow-a3dd7f50908bc5fa.rlib,libarrow_row-42f46076f0a8202c.rlib,libarrow_json-f6a2e26a7fa91775.rlib,libsimdutf8-c86ce21d8aeb7c4d.rlib,libindexmap-1628885b6d624b36.rlib,libarrow_ipc-2599808b7fd0ef0f.rlib,libzstd-f1a538d67c6f63c4.rlib,libzstd_safe-1b7bea79e9f69154.rlib,libzstd_sys-e9140d33ae3510c5.rlib,liblz4_flex-146be9587f3bf8ec.rlib,libtwox_hash-639d893ea37015d8.rlib,libflatbuffers-f3f15b60f8ebc582.rlib,libarrow_csv-629025505d34e76e.rlib,libcsv-a148a602699cbcbc.rlib,libcsv_core-16756372d163a5cd.rlib,libarrow_string-eefab598c06ba281.rlib,libregex-3cf4c9916632c5e9.rlib,libregex_automata-bcba0704fcdf863d.rlib,libaho_corasick-fc3b4750af9a3974.rlib,libregex_syntax-9ef3fb25c3f741ad.rlib,libarrow_cast-a4a9d4433c8822e6.rlib,libatoi-d562e744dc0f861f.rlib,libbase64-35f66ef5368f1397.rlib,libcomfy_table-c8caf4ecf3e707c2.rlib,libunicode_segmentation-2f8d0603ba154852.rlib,libunicode_width-c90a87ff6c559801.rlib,liblexical_core-e5924b624b5d88a4.rlib,liblexical_write_float-f1b0c07ba3325a6d.rlib,liblexical_write_integer-f1cb89b548368492.rlib,liblexical_parse_float-8b277af42d525385.rlib,liblexical_parse_integer-bfabf95e35c9109a.rlib,liblexical_util-4a5d772d17a726da.rlib,libstatic_assertions-616ecfdff0f7db80.rlib,libarrow_arith-a5dd2d05e6abfded.rlib,libarrow_ord-8ffe2e7cebaaa7d5.rlib,libarrow_select-0ca1a8dedbd82dcc.rlib,libarrow_array-f10de10ad56ed416.rlib,libchrono_tz-2c294ffa36332b7e.rlib,libphf-15fba4975b3eb0fd.rlib,libphf_shared-e2d323284e50a78b.rlib,libsiphasher-e2708e5e083f2a40.rlib,libahash-ec00cf2923410bc8.rlib,libgetrandom-c6d02c18467eb166.rlib,liblibc-25eea8833d1957b1.rlib,libonce_cell-27447345f90da58e.rlib,libzerocopy-a939f8a045416859.rlib,libhashbrown-73d80d46d1026382.rlib,libfoldhash-a64075c6f775a8f8.rlib,libequivalent-d66fda23273ff194.rlib,liballocator_api2-b9ec1af81b743a54.rlib,libchrono-2fb05a67a8185735.rlib,libiana_time_zone-565d51671a966a2b.rlib,libarrow_data-755d6053288d0f84.rlib,libarrow_schema-659a4b4426ebb20d.rlib,libserde_json-c533260cf2138d91.rlib,libmemchr-69bde76c872bf97d.rlib,libitoa-4b70524f9f3af9b2.rlib,libryu-3f9d25dbf3b0862a.rlib,libbitflags-a171966890ed4ede.rlib,libserde-4d44430c676c01b8.rlib,libarrow_buffer-c408ee0b5b4fc3f2.rlib,libbytes-68e20d69ec5ad379.rlib,libhalf-44f71933076758ff.rlib,libcfg_if-c2ea77843ae8b071.rlib,libnum-b60a65e8371b1e27.rlib,libnum_iter-5e9243854d942381.rlib,libnum_rational-f5d937c6d8a08d9c.rlib,libnum_complex-edd3ebf7e20c945c.rlib,libnum_bigint-fe7fa4e9d3d7ac13.rlib,libnum_integer-f91b773b44d3bc1e.rlib,libnum_traits-d1e12bd9e015bc5a.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,librustc_std_workspace_core-*,liballoc-*,libcore-*,libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-lbz2" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/tmp/rustc2UPVcs/raw-dylibs" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/__w/datafusion/datafusion/target/ci/build/zstd-sys-ce9d16657e01a7ff/out" "-L" "/__w/datafusion/datafusion/target/ci/build/ring-984bcf3bf6b3ebdb/out" "-L" "/__w/datafusion/datafusion/target/ci/build/psm-e260d958ca872b0f/out" "-L" "/__w/datafusion/datafusion/target/ci/build/lzma-sys-bce51c88d095743a/out" "-L" "/__w/datafusion/datafusion/target/ci/build/blake3-86fa68cee770f729/out" "-L" "/__w/datafusion/datafusion/target/ci/build/blake3-86fa68cee770f729/out" "-L" "/__w/datafusion/datafusion/target/ci/build/libmimalloc-sys-1a2475acef874fc8/out" "-L" "/usr/lib" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/__w/datafusion/datafusion/target/ci/examples/date_time_functions-ea9b6d4121c7029e" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,--strip-all" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: /usr/bin/ld: final link failed: No space left on device
          collect2: error: ld returned 1 exit status
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -347,7 +347,7 @@ jobs:
         with:
           save-if: ${{ github.ref_name == 'main' }}
           shared-key: "amd-ci-linux-test-example"
-      - name: Remove unnecessary files
+      - name: Remove unnecessary preinstalled software
         run: |
           echo "Disk space before cleanup:"
           df -h

